src/common/resolve.js

```diff
@@ -2,11 +2,16 @@
 
 let { resolve } = eval("require");
```

Comment:
```suggestion
let resolve = eval("require").resolve;
```

Reply:
I've tried this,  won't work , this will replaced by babel https://github.com/prettier/prettier/blob/master/scripts/build/babel-plugins/transform-custom-require.js
to `require.resolve` , then rollup replace it to `commonJsRequire.resolve`

Reply:
I'm not sure, but `global.require.resolve` seems a  good solution.

Reply:
Are you sure? I tested it too locally and it stays this way in `dist/index.js`:
```js
let resolve$1 = require.revolve; // In the VS Code and Atom extension `require` is overridden and `require.resolve` doesn't support the 2nd argument.

if (resolve$1.length === 1 || process.env.PRETTIER_FALLBACK_RESOLVE) {
```

Reply:
`global.require` exists only in the REPL.

Reply:
Strangely. locally ESLint doesn't issue a warning for this line for me. 🤔

Reply:
Hmm, I guess I tried `let { resolve } = require;` and`let resolve = eval("require").resolve;`. 

Eslint didn't warn on my machine either, we need add comments and disable the rule on it anyway.